### PR TITLE
WIP: add jeager service monitor

### PR DESCRIPTION
**Pull request by krasi-georgiev**
Signed-off-by: Krasi Georgiev <8903888+krasi-georgiev@users.noreply.github.com>

### CODE REVIEW

File: components/jaeger-collector.libsonnet [hunk 1]
```diff
@@ -7,11 +7,16 @@ local jaegerAgent = import './jaeger-agent.libsonnet';
     local j = self,
     namespace:: error 'must set namespace for jaeger',
     image:: error 'must set image for jaeger',
+    name:: error 'must set name for jaeger',
     replicas:: 1,
     pvc:: {
       class: 'standard',
       size: '50Gi',
     },
+    commonLabels:: {
```
Comment (brancz) [inline components/jaeger-collector.libsonnet:16]: this should also be used in the deployment

File: components/jaeger-collector.libsonnet [hunk 2]
```diff
@@ -3,110 +3,150 @@ local service = k.core.v1.service;
 local jaegerAgent = import './jaeger-agent.libsonnet';
 
 {
-  jaeger+:: {
-    local j = self,
+  local _self = self,
```
Comment (brancz) [inline components/jaeger-collector.libsonnet:6]: in other places we call this what the object is called, which would be `jaeger` in this case.. I find `_self` to be confusing as `self` is relative to the object but `_self` wouldn't be.

Comment (krasi-georgiev) [reply components/jaeger-collector.libsonnet:6]: why `_self` isn't relative to the object?

I find it confusing the other way around. for example, in many places, I see `obs` and it is very difficult to trace down what does it point to.

Comment (brancz) [reply components/jaeger-collector.libsonnet:6]: it's always the top level object that's being exported, I think the pattern is discussable, but if we change it we should do it consistently and not do it in one place but not the rest

Comment (krasi-georgiev) [reply components/jaeger-collector.libsonnet:6]: yes I see what you mean, I am still trying different things so let me think about this a bit again.

Comment (brancz) [reply components/jaeger-collector.libsonnet:6]: (if I wasn't clear I think we should not change the style for all others in this PR as well, in this PR we should stay consistent with the rest and then in a follow up we can discuss style changes for everything)

**Comment by krasi-georgiev** [comment]
will try using functions

**Pull request closed without merging**